src/Simulation/Simulators/SparseSimulator/SparseSimulatorCS/Probes.cs

```diff
@@ -0,0 +1,360 @@
+                return sim.GetAmplitude(args.Item1, args.Item2);
+            };
+
+        }
+    }
+
+    partial class GetAmplitude
+    {
+        public class Native : GetAmplitude
+        {
+            private SparseSimulator sim = null;
+            public Native(IOperationFactory m) : base(m)
+            {
+                sim = m as SparseSimulator;
+            }
+
+            public override Func<(IQArray<Qubit>, System.Numerics.BigInteger), Microsoft.Quantum.Math.Complex> __Body__ =>  args => {
+                return sim.GetAmplitude(args.Item1, args.Item2);
+            };
+
+        }
+    }
+    public partial class SparseSimulator : QuantumProcessorDispatcher, IDisposable
+    {
+        // // Converts a big integer into a string label
+        // where the label is read over the qubits passed as an argument, in order
+        // The label for all other qubits is assumed to be 0
+        // That is: Suppose we have 11 = 1011 on qubits 0, 6,9,3.
+        // That will output the string 0001001001
+        public static string InterleaveLabelByArray(IQArray<Qubit> qubits, System.Numerics.BigInteger label)
+        {
+            // Converts a big integer into a string label
+            // where the label is read over the qubits passed as an argument, in order
+            // The label for all other qubits is assumed to be 0
+            char[] newlabel = "".PadRight((int)qubits.GetIds().Max() + 1, '0').ToCharArray();
+            string test = new string(newlabel);
+            for (int idx = 0; idx < (int)qubits.Length; idx++)
+            {
+                if (((label >> idx) & 1) == 1)
+                {
+                    int id = qubits[idx].Id;
+                    newlabel[id] = '1';
+                }
+            }
+            // std::bitset parses strings backwards
+            Array.Reverse(newlabel);
+            return new string(newlabel);
+        }
+
+        //  Returns the amplitude of a single state
+        public Microsoft.Quantum.Math.Complex GetAmplitude(string label)
+        {
+            return ((SparseSimulatorProcessor)this.QuantumProcessor).GetAmplitude(label);
+        }
+        public Microsoft.Quantum.Math.Complex GetAmplitude(uint label)
+        {
+            return GetAmplitude(label.ToString());
+        }
+        public Microsoft.Quantum.Math.Complex GetAmplitude(System.Numerics.BigInteger label)
+        {
+            return GetAmplitude(label.ToString());
+        }
+
+        public Microsoft.Quantum.Math.Complex GetAmplitude(IQArray<Qubit> qubits, System.Numerics.BigInteger label)
+        {
+            return GetAmplitude(InterleaveLabelByArray(qubits, label));
+        }
+
+
+        // Dumps the state in qubits to `target` (a filename), if the qubits are 
+        // either null or not entangled to the rest of the machine;
+        // prints to console if `target` is null
+        protected virtual QVoid Dump<T>(T target, IQArray<Qubit> qubits = null)
```

Comment:
It looks like this method still uses the old way of writing out dumps to the console directly via `Message` rather than using `SimulatorBase.MaybeDisplayDiagnostic`. Using the latter with a new type that implements `object.ToString` allows IQ# and Python callers to capture diagnostics in a structured fashion rather than relying on parsing the text written to the console.

Please let me know if I can help, but we'll need to make sure this method is updated to be compatible with IQ# before merging to `main`. Thanks!